Source/CombatExtended/CombatExtended/Verbs/Verb_LaunchProjectileCE.cs

```diff
@@ -348,9 +359,128 @@ public virtual void ShiftTarget(ShiftVecReport report, bool calculateMechanicalO
                     }
                     else if (currentTarget.Thing is Pawn)
                     {
-                        // Aim for center of mass on an exposed target
-                        targetRange.min = victimVert.BottomHeight;
+                        targetRange.min = victimVert.BottomHeight;
                         targetRange.max = victimVert.MiddleHeight;
+
+                        bool AppropiateAimMode = CompFireModes?.CurrentAimMode != AimMode.SuppressFire;
+
+                        bool flagTargetMode = (CompFireModes?.targetMode != TargettingMode.automatic);
+
+                        bool isPlayerFaction = (CasterPawn?.Faction ?? Caster.Faction) == Faction.OfPlayer;
+
+                        bool flag1 = ((ShootingAccuracy >= 2.2f) | isTurret);
+
+                        if (flag1)
+                        {
+                            if (isPlayerFaction && flagTargetMode)
+                            {
+                                if (AppropiateAimMode)
+                                {
+                                    if (CompFireModes?.targetMode == TargettingMode.head)
+                                    {
+                                        // Aim upper thoraxic to head
+                                        targetRange.min = victimVert.MiddleHeight;
+                                        targetRange.max = victimVert.Max;
+                                    }
+                                    else if (CompFireModes?.targetMode == TargettingMode.legs)
+                                    {
+                                        // Aim legs to lower abdomen
+                                        targetRange.min = victimVert.Min;
+                                        targetRange.max = victimVert.BottomHeight;
+                                    }
+                                    else
+                                    {
+                                        // Aim center mass
+                                        targetRange.min = victimVert.BottomHeight;
+                                        targetRange.max = victimVert.MiddleHeight;
+                                    }
+                                }
+                            }
+                        }
+                        if (!flagTargetMode | !isPlayerFaction)

```

Comment:
I may be misunderstanding something here.  Why isn't this inside the same `if (flag1)` block?  Are you thinking enemy turrets should *not* use called shots?  

If we allow called shots for non-player turrets, or check the turret ownership when setting `IsTurret`, then this can get moved inside the block on line 399 and become a simple `else`.